core/vm/actor/actor.cpp

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "vm/actor/actor.hpp"
+
+#include <string>
+#include <vector>
+
+#include <libp2p/crypto/sha/sha256.hpp>
+
+namespace fc::vm::actor {
+  bool isBuiltinActor(const ContentIdentifier &code) {
+    return code == kStorageMarketCodeCid || code == kStoragePowerCodeCid
+           || code == kStorageMinerCodeCid || code == kAccountCodeCid
+           || code == kInitCodeCid || code == kMultisigCodeCid
+           || code == kPaymentChannelCodeCid;
+  }
+
+  bool isSingletonActor(const ContentIdentifier &code) {
+    return code == kStoragePowerCodeCid || code == kStorageMarketCodeCid
+           || code == kInitCodeCid || code == kCronCodeCid;
+  }
+
+  ContentIdentifier kEmptyObjectCid{
+      ContentIdentifier::Version::V1,
+      libp2p::multi::MulticodecType::Code::DAG_CBOR,
+      libp2p::multi::Multihash::create(libp2p::multi::HashType::sha256,
+                                       libp2p::crypto::sha256({"\xA0", 1}))
+          .value()};
+
+  ContentIdentifier makeRawIdentityCid(const std::string &str) {
+    std::vector<uint8_t> bytes{str.begin(), str.end()};
+    return {ContentIdentifier::Version::V1,
+            libp2p::multi::MulticodecType::Code::RAW,
+            libp2p::multi::Multihash::create(libp2p::multi::HashType::identity,
+                                             bytes)
+                .value()};
+  }
+
+  ContentIdentifier kAccountCodeCid = makeRawIdentityCid("fil/1/account");
```

Comment:
Should these not be constants?

Reply:
`ContentIdentifier` initialization is not constant, for example `Multihash::create` uses outcome, so we cannot use `contstexpr`